qiskit/dagcircuit/collect_blocks.py

```diff
@@ -254,22 +255,37 @@ def collapse_to_operation(self, blocks, collapse_fn):
         then uses collapse_fn to collapse this circuit into a single operation.
         """
         global_index_map = {wire: idx for idx, wire in enumerate(self.dag.qubits)}
+        global_index_map.update({wire: idx for idx, wire in enumerate(self.dag.clbits)})
+
         for block in blocks:
-            # Find the set of qubits used in this block (which might be much smaller than
-            # the set of all qubits).
+            # Find the sets of qubits/clbits used in this block (which might be much smaller
+            # than the set of all qubits/clbits).
             cur_qubits = set()
+            cur_clbits = set()
+
             for node in block:
                 cur_qubits.update(node.qargs)
+                cur_clbits.update(_get_node_cargs(node))
```

Comment:
```suggestion
                cur_clbits.update(node.cargs)
                cond = getattr(node.op, "condition", None)
                if cond is not None:
                    cur_clbits.update(condition_bits(cond))
```

Then you can remove the `_get_node_cargs` function

Reply:
Sorry for the delay on this. Initially I have thought that having an additional utility function would be helpful, but I agree that there is no need for that since it's only used once. Fixed in [db71bbe](https://github.com/Qiskit/qiskit-terra/pull/9823/commits/db71bbe961bcb967865a01af313d44b6bfa23542).